tests/python/contrib/test_hexagon/test_softmax_slice.py

```diff
@@ -0,0 +1,140 @@
+
+
+@tvm.testing.fixture
+def transformed_expected_output_np(expected_output_np, output_layout):
+    return transform_numpy(expected_output_np, output_layout)
+
+
+@tvm.testing.fixture
+def transformed_input_np(input_np, input_layout):
+    return transform_numpy(input_np, input_layout)
+
+
+class Basesoftmax2d:
+
+    input_shape, input_layout, output_layout, axis_sep = tvm.testing.parameters(
+        ((1, 1024), "nc-512c-2d", "nc-512c-2d", [2])
+    )
+    dtype = tvm.testing.parameter("float32")
+    working_scope = tvm.testing.parameter("global.vtcm")
+
+
+class TestSoftmax2d(Basesoftmax2d):
+    @tvm.testing.fixture
+    def expected_output_np(self, input_np):
+        if len(input_np.shape) == 2:
+            ref_np_2d = tvm.topi.testing.softmax_python(input_np)
+            return ref_np_2d
+        raise RuntimeError(f"Unexpected input shape '{input_np.shape}'")
+
+    @tvm.testing.requires_hexagon
+    def test_softmax_f32(
+        self,
+        dtype,
+        input_layout,
+        output_layout,
+        input_shape,
+        input_np,
+        transformed_input_np,
+        transformed_expected_output_np,
+        expected_output_np,
+        working_scope,
+        axis_sep,
+        hexagon_session,
+    ):
+
+        target_hexagon = tvm.target.hexagon(
+            "v69",
+            llvm_options="--disable-loop-unrolling-pass",
+        )
+        A = te.placeholder(input_shape, name="A", dtype=dtype)
+
+        O = sl.softmax_compute(A)
+
+        if input_layout == "nc-512c-2d":
+            tir_s = sl.softmax_stir_schedule(O, A, output_layout, input_layout)
+            sch = tir_s.mod
+        else:
+            raise RuntimeError(f"Unexpected input layout '{input_layout}'")
+
+        with tvm.transform.PassContext(
+            opt_level=3,
+            config={
+                "tir.LoopPartition": {"partition_const_loop": True},
+            },
+        ):
+
+            func = tvm.build(
+                sch,
+                [A, O],
+                tvm.target.Target(target_hexagon, host=target_hexagon),
+                name="softmax_slice",
+            )
+
+        input_arr = allocate_hexagon_array(
+            hexagon_session.device,
+            data=transformed_input_np,
+            axis_separators=axis_sep,
+            mem_scope=working_scope,
+        )
+
+        output_arr = allocate_hexagon_array(
+            hexagon_session.device,
+            tensor_shape=transformed_expected_output_np.shape,
+            dtype=transformed_expected_output_np.dtype,
+            axis_separators=axis_sep,
+            mem_scope=working_scope,
+        )
+
+        mod = hexagon_session.load_module(func)
+        mod(input_arr, output_arr)
+
+        n, c = input_np.shape
+        output_np = output_arr.numpy().reshape(1, c // 512, 512)
+
+        np.testing.assert_allclose(output_np, transformed_expected_output_np, rtol=1e-4, atol=1e-4)
+
+
+if __name__ == "__main__":
+
+    sys.exit(pytest.main(sys.argv))
```

Comment:
we can do this in a follow up:
1. remove the new line
2. replace `sys.exit(pytest.main(sys.argv))` `with tvm.testing.main()`